modules/reindex/src/main/java/org/elasticsearch/index/reindex/StartReindexJobAction.java

```diff
@@ -95,45 +95,55 @@ public boolean getWaitForCompletion() {
     public static class Response extends ActionResponse {
 
         static final ParseField TASK_ID = new ParseField("task_id");
+        static final ParseField PERSISTENT_TASK_ID = new ParseField("persistent_task_id");
         static final ParseField REINDEX_RESPONSE = new ParseField("reindex_response");
 
         private static final ConstructingObjectParser<Response, Void> PARSER = new ConstructingObjectParser<>(
-            "start_reindex_response", true, args -> new Response((String) args[0], (BulkByScrollResponse) args[1]));
+            "start_reindex_response", true, args -> new Response((String) args[0], (String) args[1], (BulkByScrollResponse) args[2]));
 
         static {
             PARSER.declareString(ConstructingObjectParser.constructorArg(), TASK_ID);
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), PERSISTENT_TASK_ID);
             PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(),
                 (parser, context) -> BulkByScrollResponse.fromXContent(parser), REINDEX_RESPONSE);
         }
 
         private final String taskId;
+        private final String persistentTaskId;
         @Nullable private final BulkByScrollResponse reindexResponse;
 
-        public Response(String taskId) {
-            this(taskId, null);
+        public Response(String taskId, String persistentTaskId) {
+            this(taskId, persistentTaskId, null);
         }
 
-        public Response(String taskId, BulkByScrollResponse reindexResponse) {
+        public Response(String taskId, String persistentTaskId, BulkByScrollResponse reindexResponse) {
             this.taskId = taskId;
+            this.persistentTaskId = persistentTaskId;
             this.reindexResponse = reindexResponse;
         }
 
         public Response(StreamInput in) throws IOException {
             super(in);
             taskId = in.readString();
+            persistentTaskId = in.readString();
             reindexResponse = in.readOptionalWriteable(BulkByScrollResponse::new);
         }
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
             out.writeString(taskId);
+            out.writeString(persistentTaskId);
             out.writeOptionalWriteable(reindexResponse);
         }
 
         public String getTaskId() {
             return taskId;
         }
 
+        public String getPersistentTaskId() {
```

Comment:
I was just working on this on a different branch and I feel like the persistent task id should be the default naming? As in the methods are `getTaskId` or  `getReindexTaskId`? And the other one be `getEphemeralTaskId`? Or `getCoordinatorNodeTaskId`? Or `getInitialEphemeralTaskId`?